src/Extensions/BaseElementExtension.php

```diff
@@ -2,231 +2,357 @@
+    }
+
+    /***
+     * @return void
+     */
+    public function populateDefaults()
+    {
+        $defaultSizeField = 'Size' . ElementalConfig::getDefaultViewport();
+        $this->owner->$defaultSizeField = self::GRID_COLUMNS_COUNT;
+    }
 
     /**
      * @var array
      */
-    private static $db = array(
+    private static $titleOptions = [
+        'div' => 'Default',
+        'h1' => 'H1',
+        'h2' => 'H2',
+        'h3' => 'H3',
+        'h4' => 'H4',
+        'h5' => 'H5',
+        'h6' => 'H6',
+    ];
+
+    /**
+     * @var array
+     */
+    private static $db = [
         'SizeXS' => 'Int',
         'SizeSM' => 'Int',
         'SizeMD' => 'Int',
         'SizeLG' => 'Int',
-
+        'SizeXL' => 'Int',
         'OffsetXS' => 'Int',
         'OffsetSM' => 'Int',
         'OffsetMD' => 'Int',
         'OffsetLG' => 'Int',
+        'OffsetXL' => 'Int',
+        'VisibilityXS' => 'Varchar(10)',
+        'VisibilitySM' => 'Varchar(10)',
+        'VisibilityMD' => 'Varchar(10)',
+        'VisibilityLG' => 'Varchar(10)',
+        'VisibilityXL' => 'Varchar(10)',
+        'TitleClass' => 'Varchar(100)',
+        'TitleTag' => 'Varchar(100)',
+    ];
 
-        'VisibilityXS' => 'Varchar',
-        'VisibilitySM' => 'Varchar',
-        'VisibilityMD' => 'Varchar',
-        'VisibilityLG' => 'Varchar',
+    /***
+     * @var array
+     */
+    private static $defaults = [
+        'ShowTitle' => true,
+    ];
 
-        'BlockType' => 'Varchar'
-    );
+    /***
+     * @return array
+     */
+    public function getTitleHTMLTags()
+    {
+        return self::$titleOptions;
+    }
 
-    public function populateDefaults()
+    /***
+     * @return array
+     */
+    public function getTitleClasses()
     {
-        $defaultSizeField = 'Size' . Config::forClass('TheWebmen\ElementalGrid')->get('defaultSizeField');
-        $this->owner->$defaultSizeField = Config::forClass('TheWebmen\ElementalGrid\Extensions\BaseElementExtension')->get('default_col_size');
+        $classes = self::$titleOptions;
+
+        $this->owner->extend('updateTitleClasses', $classes);
+
+        return $classes;
     }
 
     /**
-     * Get the options for col sizes
-     * @return array
+     * Update the default CMS fields with our custom fields
+     *
+     * @param FieldList $fields
      */
-    public static function getColSizeOptions($includeDefault = false, $includeNone = false){
-        $config = Config::inst()->get(__CLASS__);
-        $numColumns = $config['num_columns'];
-        $out = array();
-        if($includeDefault){
-            $out[0] = _t(__CLASS__ . '.DEFAULT', 'Default');
-        }else if($includeNone){
-            $out[0] = _t(__CLASS__ . '.NONE', 'None');
-        }
-        for($i = 1; $i < $numColumns + 1; $i++){
-            $out[$i] = _t(__CLASS__ . '.COLUMN', 'Column') . ' ' . $i . '/' . $numColumns;
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->removeByName('Title');
```

Comment:
This `$fields->removeByName` linkes can be replaced by 1 single line;
`$fields->removeByName(['Title', 'TitleClass', 'TitleTag']);`